### ISSUE TITLE
Some vertical polygons rendering incorrectly

### ISSUE THREAD

**Issue by JacksonBates** [AnalyticalGraphicsInc/cesium#6855]
Continuing from https://github.com/AnalyticalGraphicsInc/cesium/issues/6662 after the 1.47 update.

The vertical pentagon mentioned in the previous issue by @silky still does not render correctly after the PR and update, but many similar vertical pentagons do (the important relevant cases are the **pink** and **orange** polygons):

<img width="349" alt="screen shot 2018-07-30 at 12 33 06 pm" src="https://user-images.githubusercontent.com/4337740/43374900-54868a5e-93f5-11e8-865b-5b70eefd4bfc.png">

- The green example is our base case which works: A symmetrical, vertical pentagon.
- The red example breaks as the vertices are listed 'out of order', i.e. they do not follow the route either clockwise or anti-clockwise.
- The blue works, as an example of assymmetry.
- The orange is broken - the only difference between this and blue is that the base is a little wider
- The pink works - the only difference between this and orange is that the high mid-point is shifted a little.

Here is the relevant sandcastle demo, including inline comments: [Demo](https://cesiumjs.org/Cesium/Apps/Sandcastle/index.html#c=zZbdb9owEMD/lRMvBYmar1KtHWVru6qrNkHVdnsZezDJQSwcG9kOKJv2v+8cU8ooUUefaqKI3Ffs+93ZWXADC4FLNHAGCpdwiVZkKfteyKoHUfF4qZXjQqE5qL0fKf9rNOCep3OJdVhqMxNqCgs0TkRcwlzLfKqVHakFRZ8aRHUbRPSO8DKGygkn0DIex9XfIwU0FE8RTuHg2ns8RlmHlflBPdg9ak5h5VgI0dxqSzG1+oximrhTcCaj6a3Gk2Ui0HATJTkFWK32khtH/7jqsInR6Sf0k7bnxvA8BLPVH+sArVazDp1WHdpNP+obilZQtMoUm7LmSblsO8DPWn1jASl3aAQl+mn+WmrDru+urgZsKVxyLucJrzbZcS14/RmpP4EccRtoh5ScBEGbmLjrySrHaH1oim3BG1bTzDoYI0REXyidWUCKTR6R1NFsKayPAJyUh2tJLUA3GO+F/A7jtwzc42uXcW3/y7CojvaO6ijgtncAb78W+N3Vp5dxn1ubp+hMDhYxtaBnAdFYZrgXowtyeNNdybqUzE5ZX3bK+HV2ICmRvQ7TxddvVy9zusgcWE2p9phoP5U5cDMWjrKSQywmEzSofI9ygzAYPsDwSwCpDVfT/VAOC5e3vsUelaE8KkN5tIXt3aMsCHw3gBTOSYSloL3vuekqrjfdk/Lw7nxw/R+cHxJhgS5O11gvKH1jIu/dij05Im7GoxGK6iERE0dbI5V2YD0XarYX6VtyePtN212TLhjZgk49LH+VkS3S3bIS6O4ogRLZ67r59mbw5WXKNxNY4moBLuGOUjVNkM7TAHacA+EU/kzlqc6Us4UXuUzRrbvdn74JX5CZsaeFwQc/DmkFrAtw2Icb5VDF/vAka04kCyuvPyarEz8tsmoTPyIylWhX+hPW8np2Uuhba31QExVS0/1ot7undlhMpWwSsDGockOd/tI6fdDVraqtva/UKz3rcon94PhRpHNtHGRGVhlrOKTvTIJhG+MsmqFjkbU+z73Go1MvFgsQ8dmosvW5OqrQtwq3ljSTTMp78QtHlX6vQfb/uEnNY9pyh9QgkufeJGn1vwYhY6zXoMfnXk5rOeZmI+Jf)

The pink polygon is followed by some comments specifying ranges of coordinates for that high mid-point. Some render the pentagon normally, some present it as two triangles, and some present it as one triangle.

Code from the demo, for quick reference:

```js
var viewer = new Cesium.Viewer('cesiumContainer');


// Sample, working vertical polygons
var greenPolygon = viewer.entities.add({
    name : 'Green polygon vertically',
    polygon : {
        perPositionHeight: true,        
        hierarchy : Cesium.Cartesian3.fromDegreesArrayHeights([
       110, 31, 200000,
       111, 31, 100000,
       111, 31, 0,
       109, 31, 0,
       109, 31, 100000,
       ]), 
        material : Cesium.Color.GREEN.withAlpha(0.6)
    }
});

// Note: the order of vertices matters 
//(must be continous either clockwise or anti-clockwise)
var redPolygon = viewer.entities.add({
    name : 'Red polygon vertically',
    polygon : {
        perPositionHeight: true,        
        hierarchy : Cesium.Cartesian3.fromDegreesArrayHeights([
       111, 32, 100000,
       111, 32, 0,
       110, 32, 200000,
       109, 32, 0,
       109, 32, 100000,
       ]), 
        material : Cesium.Color.RED.withAlpha(0.6)
    }
});

// Assymetry seems ok
var bluePolygon = viewer.entities.add({
    name : 'Blue polygon vertically',
    polygon : {
        perPositionHeight: true,        
        hierarchy : Cesium.Cartesian3.fromDegreesArrayHeights([
       110.5, 33, 200000,
       111, 33, 100000,
       111, 33, 0,
       109, 33, 0,
       109, 33, 100000,
       ]), 
        material : Cesium.Color.BLUE.withAlpha(0.6)
    }
});

// But some seemingly arbitrary differences are NOT OK
var orangePolygon = viewer.entities.add({
    name : 'Orange polygon vertically',
    polygon : {
        perPositionHeight: true,        
        hierarchy : Cesium.Cartesian3.fromDegreesArrayHeights([
       110, 34, 200000,
       111, 34, 100000,
       111, 34, 0,
       108, 34, 0,      // A little wider
       108, 34, 100000, // 
       ]), 
        material : Cesium.Color.ORANGE.withAlpha(0.6)
    }
});

// This is as above, but with the centre point shifted 0.5
var pinkPolygon = viewer.entities.add({
    name : 'Pink polygon vertically',
    polygon : {
        perPositionHeight: true,        
        hierarchy : Cesium.Cartesian3.fromDegreesArrayHeights([
       110.5, 35, 200000, // As wide, shift centre
       111, 35, 100000,
       111, 35, 0,
       108, 35, 0,
       108, 35, 100000,
       ]), 
        material : Cesium.Color.PINK.withAlpha(0.6)
    }
});

// If we shift that highest point by various amounts
// we get different behaviours:
// ?????-108.5  -> Intended pentagon
// 108.6-109    -> 2 triangles
// 109.1-109.9  -> 1 triangle
// 110.1-110.4  -> 2 triangles
// 110.5-?????  -> Intended pentagon
            
viewer.zoomTo(viewer.entities);
```

**Comment by hpinkos** [comment]
Thanks @JacksonBates!  I'll look into this today.  The positions of the polygon must be listed in order, so the red one not working is expected and isn't something we'll fix.  But the issue with orange polygon seems to be a bug.

**Comment by hpinkos** [comment]
Actually, nevermind.  It looks like you were using Cesium 1.47.  Vertical polygon support is in master but hasn't been included in a release yet.  It will be included in the 1.48 release available on August 1st.   Here's  preview:

![image](https://user-images.githubusercontent.com/3451886/43403979-6d11f614-93e4-11e8-9a9e-362b56e4e8f4.png)

**Comment by JacksonBates** [comment]
This is great, thanks @hpinkos!
